tests/foreman/cli/test_satellitesync.py

```diff
@@ -763,6 +807,382 @@ def test_negative_export_cv_with_on_demand_repo(self):
             'All exported repositories must be set to an immediate download policy and re-synced'
         )
 
+    @tier1
+    def test_negative_export_cv_with_background_policy_repo(self):
+        """Exporting CV version having background policy repo throws error
+
+        :id: c0f5a903-e9a8-4ce6-9377-1df1e7ba62c5
+
+        :steps:
+
+            1. Create product and background policy repository with custom contents
+            2. Sync the repository
+            3. Create CV with above product and publish
+            4. Attempt to export CV version contents to a directory
+
+        :expectedresults:
+
+            1. Export fails with error 'All exported repositories must be set to an immediate
+                download policy and re-synced' should be displayed.
+
+        """
+        exporting_org = make_org()
+        exporting_prod = gen_string('alpha')
+        product = make_product({
+            'organization-id': exporting_org['id'],
+            'name': exporting_prod
+        })
+        repo = make_repository({
+            'name': gen_string('alpha'),
+            'download-policy': 'background',
+            'product-id': product['id']
+        })
+        Repository.synchronize({'id': repo['id']})
+        _, exporting_cvv_id = ContentViewSync._create_cv(
+            gen_string('alpha'), repo, exporting_org)
+        with self.assertRaises(CLIReturnCodeError) as error:
+            ContentView.version_export({
+                'export-dir': '{}'.format(self.export_base),
+                'id': exporting_cvv_id
+            })
+        self.assert_error_msg(
+            error,
+            'All exported repositories must be set to an immediate download policy and re-synced'
+        )
+
+    @tier1
+    def test_negative_import_cv_with_mirroronsync_repo(self):
+        """Importing CV version having mirror-on-sync repo throws error
+
+        :id: dfa0cd5f-1596-4097-b505-06bc14de51dd
+
+        :steps:
+
+            1. Create product and mirror-on-sync repository with custom contents
+            2. Sync the repository
+            3. Create CV with above product and publish
+            4. Attempt to export CV version contents to a directory
```

Comment:
are you trying to import or export here?

Reply:
Export.

We have to export it to the directory only. And we can import from directory only. In both export and import, directory comes into the pic.